records_mover/url/optimizer/__init__.py

```diff
@@ -0,0 +1,85 @@
+from contextlib import contextmanager
+from records_mover.url.base import BaseDirectoryUrl
+from records_mover.url.gcs.gcs_directory_url import GCSDirectoryUrl
+from records_mover.url.s3.s3_directory_url import S3DirectoryUrl
+from typing import Iterator, Tuple, Union, Optional
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+class CopyOptimizer:
+    # TODO: move to new location
+    # TODO: Bring in Google code
+    # TODO: Figure out where this should be called from
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker